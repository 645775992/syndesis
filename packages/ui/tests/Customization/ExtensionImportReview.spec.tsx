import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { fireEvent, render } from 'react-testing-library';
import {
  ExtensionImportReview,
  IExtensionImportReviewProps,
  IImportAction,
} from '../../src/Customization';

const mockOnImport = jest.fn();

export default describe('ExtensionImportReview', () => {
  const fourActions = [
    {
      name: 'Action 1',
      description: 'The description for action 1',
    } as IImportAction,
    {
      name: 'Action 2',
      description: 'The description for action 2',
    } as IImportAction,
    {
      name: 'Action 3',
      description: 'The description for action 3',
    } as IImportAction,
    {
      name: 'Action 4',
      description: 'The description for action 4',
    } as IImportAction,
  ] as IImportAction[];

  const oneAction = [fourActions[1]] as IImportAction[];

  const actionText = (name: string, description: string) => {
    return `${name} - ${description}`;
  };

  const cancelBtnText = 'Cancel';
  const cancelLink = '/extensions';
  const description = 'An extension to Syndesis to do Logging';
  const descriptionLabel = 'Description';
  const id = 'io.syndesis.extensions:syndesis-extension-log';
  const idLabel = 'ID';
  const importBtnText = 'Import Extension';
  const name = 'Log';
  const nameLabel = 'Name';
  const stepsActionsLabel = 'Steps';
  const title = 'Import Review';
  const typeLabel = 'Type';
  const typeMsg = 'Step Extension';

  const props = {
    actions: fourActions,
    cancelLink: cancelLink,
    extensionDescription: description,
    extensionId: id,
    extensionName: name,
    i18nActionsLabel: stepsActionsLabel,
    i18nCancel: cancelBtnText,
    i18nDescriptionLabel: descriptionLabel,
    i18nExtensionTypeMessage: typeMsg,
    i18nIdLabel: idLabel,
    i18nImport: importBtnText,
    i18nNameLabel: nameLabel,
    i18nTitle: title,
    i18nTypeLabel: typeLabel,
    i18nActionText: actionText,
    onImport: mockOnImport,
  } as IExtensionImportReviewProps;

  const { extensionDescription, ...noDescriptionProps } = props;

  beforeEach(() => {
    mockOnImport.mockReset();
  });

  it('Should render correctly with description', () => {
    const componentWithDescription = (
      <MemoryRouter>
        <ExtensionImportReview {...props} />
      </MemoryRouter>
    );

    const { getByText, queryAllByText, queryByText } = render(
      componentWithDescription
    );

    // title
    expect(queryAllByText(title)).toHaveLength(1);

    // id label
    expect(queryAllByText(idLabel)).toHaveLength(1);

    // id value
    expect(queryAllByText(id)).toHaveLength(1);

    // name label
    expect(queryAllByText(nameLabel)).toHaveLength(1);

    // name value
    expect(queryAllByText(name)).toHaveLength(1);

    // description label
    expect(queryAllByText(descriptionLabel)).toHaveLength(1);

    // description value
    expect(queryAllByText(description)).toHaveLength(1);

    // import button
    expect(queryAllByText(importBtnText)).toHaveLength(1);
    fireEvent.click(getByText(importBtnText));
    expect(mockOnImport).toHaveBeenCalledTimes(1);

    // cancel button
    expect(queryAllByText(cancelBtnText)).toHaveLength(1);
    expect(queryByText(cancelBtnText)).toHaveAttribute('href', cancelLink);

    // actions
    fourActions.map(a => {
      expect(queryByText(actionText(a.name, a.description))).toBeDefined();
    });
  });

  it('Should render correctly without a description', () => {
    const componentWithoutDescription = (
      <MemoryRouter>
        <ExtensionImportReview {...noDescriptionProps} actions={oneAction} />
      </MemoryRouter>
    );

    const { getByText, queryAllByText, queryByText } = render(
      componentWithoutDescription
    );

    // title
    expect(queryAllByText(title)).toHaveLength(1);

    // id label
    expect(queryAllByText(idLabel)).toHaveLength(1);

    // id value
    expect(queryAllByText(id)).toHaveLength(1);

    // name label
    expect(queryAllByText(nameLabel)).toHaveLength(1);

    // name value
    expect(queryAllByText(name)).toHaveLength(1);

    // description label
    expect(queryAllByText(descriptionLabel)).toHaveLength(1);

    // description value
    expect(queryByText(description)).toBeNull();

    // import button
    expect(queryAllByText(importBtnText)).toHaveLength(1);
    fireEvent.click(getByText(importBtnText));
    expect(mockOnImport).toHaveBeenCalledTimes(1);

    // cancel button
    expect(queryAllByText(cancelBtnText)).toHaveLength(1);
    expect(queryByText(cancelBtnText)).toHaveAttribute('href', cancelLink);

    // actions
    oneAction.map(a => {
      expect(queryByText(actionText(a.name, a.description))).toBeDefined();
    });
  });
});
