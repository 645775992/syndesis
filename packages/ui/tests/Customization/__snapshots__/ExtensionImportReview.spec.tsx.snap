// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtensionImportReview Should render correctly with description 1`] = `
<div
  className="extension-import-review__container container"
>
  <div
    className="extension-import-review__title row"
  >
    Import Review
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      ID
    </div>
    <div
      className="extension-import-review__propertyValue"
    >
      io.syndesis.extensions:syndesis-extension-log
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      Name
    </div>
    <div
      className="extension-import-review__propertyValue"
    >
      Log
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      Description
    </div>
    <div
      className="extension-import-review__propertyValue"
    >
      An extension to Syndesis to do Logging
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      Type
    </div>
    <div
      className="extension-import-review__propertyValue"
    >
      Step Extension
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      Steps
    </div>
    <div
      className="container-fluid"
    >
      <div
        className=""
        dangerouslySetInnerHTML={
          Object {
            "__html": "Action 1 - The description for action 1",
          }
        }
      />
      <div
        className="row"
      >
        <div
          className="col-xs-2"
        />
        <div
          className=""
          dangerouslySetInnerHTML={
            Object {
              "__html": "Action 2 - The description for action 2",
            }
          }
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-xs-2"
        />
        <div
          className=""
          dangerouslySetInnerHTML={
            Object {
              "__html": "Action 3 - The description for action 3",
            }
          }
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-xs-2"
        />
        <div
          className=""
          dangerouslySetInnerHTML={
            Object {
              "__html": "Action 4 - The description for action 4",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-offset-2"
    >
      <div
        className="row"
      >
        <div
          className=""
        >
          <div
            className="container-fluid extension-import-review__buttonBar"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Import Extension
            </button>
            <a
              className="btn btn-default extension-import-review__cancelButton"
              href="/extensions"
              onClick={[Function]}
            >
              Cancel
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ExtensionImportReview Should render correctly without a description 1`] = `
<div
  className="extension-import-review__container container"
>
  <div
    className="extension-import-review__title row"
  >
    Import Review
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      ID
    </div>
    <div
      className="extension-import-review__propertyValue"
    >
      io.syndesis.extensions:syndesis-extension-log
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      Name
    </div>
    <div
      className="extension-import-review__propertyValue"
    >
      Log
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      Description
    </div>
    <div
      className="extension-import-review__propertyValue"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      Type
    </div>
    <div
      className="extension-import-review__propertyValue"
    >
      Step Extension
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="extension-import-review__propertyLabel col-xs-2"
    >
      Steps
    </div>
    <div
      className="container-fluid"
    >
      <div
        className=""
        dangerouslySetInnerHTML={
          Object {
            "__html": "Action 2 - The description for action 2",
          }
        }
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-offset-2"
    >
      <div
        className="row"
      >
        <div
          className=""
        >
          <div
            className="container-fluid extension-import-review__buttonBar"
          >
            <button
              className="btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Import Extension
            </button>
            <a
              className="btn btn-default extension-import-review__cancelButton"
              href="/extensions"
              onClick={[Function]}
            >
              Cancel
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
